src/resizing.jl

```diff
@@ -156,34 +156,58 @@ end
 # imresize
 imresize(original::AbstractArray, dim1, dimN...) = imresize(original, (dim1,dimN...))
 
-function imresize{T,N}(original::AbstractArray{T,N}, short_size::NTuple)
+function imresize(original::AbstractArray, short_size::Tuple)
     len_short = length(short_size)
-    new_size = ntuple(i -> (i > len_short ? size(original,i) : short_size[i]), N)
+    new_size = ntuple(i -> (i > len_short ? odims(original, i, short_size) : short_size[i]), N)
```

Comment:
Also I just now realize I could have written `Val{N}` in the original code, which I didn't for some reason